pandas/tseries/tests/test_resample.py

```diff
@@ -660,6 +660,20 @@ def test_resample_empty(self):
         rs = xp.resample('A')
         assert_frame_equal(xp, rs)
 
+        # These were sometimes causing a segfault (for the functions with
+        # bounds-checking disabled) or an IndexError.  We just run them to
+        # ensure they no longer do.  (#10228)
+        index = pd.DatetimeIndex([])
```

Comment:
I would actually make a couple of loops here, iterate over all of the index types, then all of the possible dtypes, then all of the 'hows' and exhaustivly test the possibilities

I would actually define these functions in ``utils.testing`` (right below the make index) functions
```
def all_index_types():
  return [ tm.makeIntIndex, tm.makeFloatIndex,
               tm.makeStringIndex, tm.makeUnicodeIndex,
                tm.makeDateIndex, tm.makePeriodIndex, 
                tm.makeTimedeltaIndex, tm.makeBoolIndex, 
                tm.makeCategoricalIndex]

def datetimelike_index_types():
     return [  tm.makeDateIndex, tm.makePeriodIndex, 
                tm.makeTimedeltaIndex ]
```

obviously use the 2nd one